railties/lib/rails/generators/actions.rb

```diff
@@ -211,9 +211,9 @@ def initializer(filename, data = nil)
       #   generate(:authenticated, "user session")
       def generate(what, *args)
         log :generate, what
-        argument = args.flat_map(&:to_s).join(" ")
+        argument = args.flat_map(&:to_s)
 
-        in_root { run_ruby_script("bin/rails generate #{what} #{argument}", verbose: false) }
```

Comment:
Have you tried `rails` instead `bin/rails`? It always shellout and it was working and for the report you have it was failing because `bin/rails` doesn't exist in the execution directory but `rails` will work in any directory.

Reply:
I didn't investigate that much but changing the `run_ruby_script("bin/rails ...")` to `run_ruby_script("rails ...")` produces the same problem (i.e. `No such file or directory -- rails`).

On the other hand, relying on `system("rails generate ...")` properly triggers the generation but the file are placed inside the application's `app` and `test` directories rather than the specified `source_destination` in the test.

Reply:
@rafaelfranca yes, @robin850 is correct...I tried your suggestion of changing `bin/rails` to `rails`, but I got the same issue in tests. Additionally, when actually trying to run it in the real world, I got an additional "No such file or directory" error printed to STDOUT. That said, I don't believe the ability to test the results of a generator called from another generator ever worked, at least I can't remember a time when it did...